Streampipes 634 - ADD MkDocs Documentation

kulbachcedric opened the pull request:
<!--
  ~ Licensed to the Apache Software Foundation (ASF) under one or more
  ~ contributor license agreements.  See the NOTICE file distributed with
  ~ this work for additional information regarding copyright ownership.
  ~ The ASF licenses this file to You under the Apache License, Version 2.0
  ~ (the "License"); you may not use this file except in compliance with
  ~ the License.  You may obtain a copy of the License at
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~
  -->
  
  <!--
Thanks for contributing! Here are some tips you can follow to help us incorporate your contribution quickly and easily:
1. If this is your first time, please read our contributor guidelines:
    - https://streampipes.apache.org/getinvolved.html
    - https://cwiki.apache.org/confluence/display/STREAMPIPES/Getting+Started
2. Make sure the PR title is formatted like: `[STREAMPIPES-<Jira issue #>] PR title ...`
3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., `[WIP][STREAMPIPES-<Jira issue #>] PR title ...`.
4. Please write your PR title to summarize what this PR proposes/fixes.
5. Be sure to keep the PR description updated to reflect all changes.
6. If possible, provide a concise example to reproduce the issue for a faster review.
7. Make sure tests pass via `mvn clean install`.
8. (Optional) If the contribution is large, please file an Apache ICLA
    - http://apache.org/licenses/icla.pdf
-->

### Purpose

Added the MkDocs Documentation to the Python client.


### Remarks
<!--
List related issues/PRs, link to discussions in the mailing list, todo items, or any other notes related to the PR.
-->
Fixes: <Jira issue link>(https://issues.apache.org/jira/browse/STREAMPIPES-634)

bossenti commented:
@kulbachcedric thanks a lot for your initiative 🙏🏼 
The docs already look pretty cool 🌠 
Looking forward to improve it even further.

I just fixed some emojis which were not rendered correctly and added missing Apache headers.
I will rebase again once PR-142 is merged.

kulbachcedric commented:
> @kulbachcedric thanks a lot for your initiative 🙏🏼 The docs already look pretty cool 🌠 Looking forward to improve it even further.
> 
> I just fixed some emojis which were not rendered correctly and added missing Apache headers. I will rebase again once PR-142 is merged.

Nice, thanks!! 😁

bossenti commented on streampipes-client-python/setup.py:
```diff
@@ -43,7 +43,7 @@
     "isort>=5.10.1",
     "mypy>=0.990",
-    "pandas-stubs>=1.5.1.221024",
+    "pandas-stubs>=1.2.0.62",
```
@kulbachcedric Any specific reason? (just out of curiosity)

kulbachcedric replied:
> @kulbachcedric Any specific reason? (just out of curiosity)
For now, It's for python 3.11 ...

bossenti commented on streampipes-client-python/docs/getting-started/installation.md:
```diff
@@ -0,0 +1,39 @@
+*Before opening a pull request*, review the [Get Involved](https://streampipes.apache.org/getinvolved.html) page.
+It lists information that is required for contributing to StreamPipes.
+
+When you contribute code, you affirm that the contribution is your original work and that you
```
@dominikriemer is the wording here appropriate?

bossenti commented on streampipes-client-python/docs/getting-started/installation.md:
````diff
@@ -0,0 +1,39 @@
+```
+
+---
+# 👨‍💻 Contributing
````
@kulbachcedric We could additionally link to our DEVELOPMENT.md here. What do you think?

kulbachcedric replied:
Actually, I copied it from there 🙈. 
A clean link would definitely be an improvement...
